### PR TITLE
#2 saving selected in localstorage. Commit from clone

### PR DESCRIPTION
**Pull request by albertocaparros**
Pull request for issue #2 saving selected in localstorage. Working from cloned repository.

### CODE REVIEW

File: src/pods/configure-verbs/configure-verbs.container.tsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ export const ConfigureVerbsContainer = () => {
       []
     );
 
+    localStorage.setItem('selectedVerbs', JSON.stringify(selectionKeys));
```
Comment (brauliodiez) [inline src/pods/configure-verbs/configure-verbs.container.tsx:30]: Let's add a wrapper for the localStorage operations, this wrapper:
- Will take care of error handling (e.g. some broswsers in incognito mode can have disallowed accesing local storage).

- Will take care of stringyfing the values

About the Key 'selectedVerbs' we should store this in a const

File: src/pods/home/home.container.tsx [hunk 2]
```diff
@@ -10,110 +11,15 @@ export const HomeContainer = () => {
...
-      'write',
-    ]);
+    const selectedVerbs: string[] = JSON.parse(
+      localStorage.getItem('selectedVerbs')
+    );
+
+    if (selectedVerbs) verbsContext.setSelectedVerbs(selectedVerbs);
+    else {
+      verbsContext.setSelectedVerbs(initialSelected);
+      localStorage.setItem('selectedVerbs', JSON.stringify(initialSelected));
```
Comment (brauliodiez) [inline src/pods/home/home.container.tsx:21]: Same here, mi advice

- Create a localStorage wrapper
- Use this wrapper in an "api" like file, and use promises, by using this approach we could easily migrate to rest-api without having to impact the rest of the application

**Comment by albertocaparros** [comment]
Last commit: 

- Using Lockr (https://github.com/tsironis/lockr) for wrapping localstorage. Seems lightweight and intuitive for this use.
- Moving verbCollection api and vm from pods/home to core/verbs
- Adding selectedVerbCollection to said api
- Calls from home.container and configure-verbs.container now use verbs api

File: package.json [hunk 3]
```diff
@@ -40,6 +40,7 @@
     "emotion": "^10.0.27",
     "formik": "^2.1.4",
     "immer": "^7.0.9",
+    "lockr": "^0.8.5",
```
Comment (brauliodiez) [inline package.json:43]: Mmmm lockr is 4 years old (2 years without any maintanance) and it does not include typescript typings.

Should we remove this lib and use this TypeScript Gist?

https://gist.github.com/MichalZalecki/70a831304b0e83afbc0fbad3e8577068

File: src/core/verbs/global-verbs.api.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  Promise.resolve(fullVerbCollection);
+
+export const loadSelectedVerbCollection = async (): Promise<string[]> => {
+  selectedVerbCollection = Lockr.get('selectedVerbs', defaultSelectedVerbs);
+  return Promise.resolve(selectedVerbCollection);
+};
+
+export const saveFullVerbCollection = async (
+  verbCollection: VerbEntityApi[]
+) => {};
```
Comment (brauliodiez) [inline src/core/verbs/global-verbs.api.ts:28]: Mmm I guess we don't need this method,

File: src/core/verbs/global-verbs.api.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  'selectedVerbs',
+  defaultSelectedVerbs
+);
+
+// Simulating AJAX Call
+export const loadFullVerbCollection = async (): Promise<VerbEntityApi[]> =>
+  Promise.resolve(fullVerbCollection);
+
+export const loadSelectedVerbCollection = async (): Promise<string[]> => {
+  selectedVerbCollection = Lockr.get('selectedVerbs', defaultSelectedVerbs);
```
Comment (brauliodiez) [inline src/core/verbs/global-verbs.api.ts:22]: 'selectedVerbs'

This should be enclosed in a const on top of this file (We are using the same magic string in line 22 and 33, easy to introduce a typo and a hard to pinpoint bug)

**Comment by albertocaparros** [comment]
- removed Lockr
- new localstorage wrapper in /core/local-storage
- global-verbs api refactor for new wrapper and use of storage key constants

**Pull request merged**